codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/customizations/serde/SerdeDecorator.kt

```diff
@@ -26,35 +19,22 @@ val SerdeModule =
         documentationOverride = "Implementations of `serde` for model types. NOTE: These implementations are NOT used for wire serialization as part of a Smithy protocol and WILL NOT match the wire format. They are provided for convenience only.",
     )
 
-class ClientSerdeDecorator : ClientCodegenDecorator {
-    override val name: String = "ClientSerdeDecorator"
-    override val order: Byte = 0
-
-    override fun extras(
-        codegenContext: ClientCodegenContext,
-        rustCrate: RustCrate,
-    ) = extrasCommon(codegenContext, rustCrate)
-}
-
-class ServerSerdeDecorator : ServerCodegenDecorator {
-    override val name: String = "ServerSerdeDecorator"
-    override val order: Byte = 0
-
-    override fun extras(
-        codegenContext: ServerCodegenContext,
-        rustCrate: RustCrate,
-    ) = extrasCommon(codegenContext, rustCrate)
-}
-
-// Just a common function to keep things DRY.
-private fun extrasCommon(
+/**
+ * The entrypoint to both the client and server decorators.
+ */
+fun extrasCommon(
```

Comment:
can we rename this something related to serde—just because it's being called from an external context.

`serdeExtras` maybe?